opds_import.py

```diff
@@ -925,50 +879,143 @@ def __init__(self, _db, feed_url, default_data_source, import_class,
 
         Long timeout, raise error on anything but 2xx or 3xx.
         """
         kwargs = dict(timeout=120, allowed_response_codes=['2xx', '3xx'])
-        return HTTP.get_with_timeout(url, **kwargs)
+        response = HTTP.get_with_timeout(url, headers=headers, **kwargs)
+        return response.status_code, response.content_type, response.content
+
+    def check_for_new_data(self, feed, cutoff_date=None):
+        """Check if the feed contains any entries that haven't been imported yet
+        or have been updated since the cutoff date.
+        """
+        last_update_dates = self.importer.extract_last_update_dates(feed)
+
+        new_data = False
+        for identifier, updated in last_update_dates:
+
+            identifier, ignore = Identifier.parse_urn(self._db, identifier)
+            data_source = DataSource.lookup(self._db, self.importer.data_source_name)
+            record = None
+
+            if identifier:
+                record = CoverageRecord.lookup(
+                    identifier, data_source, operation=CoverageRecord.IMPORT_OPERATION
+                )
+
+            # If we have a CoverageRecord, that's the most reliable indicator of the last time we tried
+            # to import this book. But if we imported the book before we started creating CoverageRecords
+            # for imports, we can still use the monitor's timestamp as the cutoff.
+            if record:
+                cutoff = record.timestamp
+            else:
+                cutoff = cutoff_date
+
+            if cutoff:
+                # We've imported this book before, so don't import it again unless it's changed.
+
+                if not updated:
+                    # If we don't know if the book has been updated, import it again to be safe.
+                    new_data = True
+                    break
 
-    def follow_one_link(self, link, start):
-        self.log.info("Following next link: %s, cutoff=%s", link, start)
-        response = self._get(link)
+                if  updated >= cutoff:
+                    # This book has been updated.
+                    new_data = True
+                    break
 
-        imported, messages, next_links = self.importer.import_from_feed(
-            response.content, even_if_no_author=True, cutoff_date=start,
+            if record and record.exception:
+                # We tried to import this book before and it failed. The book hasn't been updated,
+                # so it probably still won't work. Don't try again.
+                continue
+
+            # There's no record of a failure for this book. Has it been imported from this source already?
+            existing_edition = get_one(
+                self._db, Edition,
+                primary_identifier=identifier,
+                data_source=data_source,
+            )
+            if not existing_edition:
+                new_data = True
+                break
```

Comment:
I think you're going to be reimporting them all anyway, because those old Editions will have their original DataSources, and the new ones will all be associated with the content server, right? Because we did those imports before we had the presentation-edition code. So I would expect existing_edition to always fail on our existing production site.